terminus/builders/osm_city_builder.py

```diff
@@ -37,35 +37,30 @@ def get_city(self):
         # Get origin of the map (we use the center)
         tree = ET.parse(self.osm_map)
         bounds = tree.find('bounds').attrib
-        self.bounds = {'minlat': float(bounds['minlat']),
-                       'maxlat': float(bounds['maxlat']),
-                       'minlon': float(bounds['minlon']),
-                       'maxlon': float(bounds['maxlon'])}
+        self.bounds = {'origin': LatLon(float(bounds['minlat']), float(bounds['minlon'])),
+                       'corner': LatLon(float(bounds['maxlat']), float(bounds['maxlon']))}
 
-        self.map_origin = LatLon((self.bounds['minlat'] + self.bounds['maxlat']) / 2,
-                                 (self.bounds['minlon'] + self.bounds['maxlon']) / 2)
+        self.map_origin = self.bounds['origin'].middle_point(self.bounds['corner'])
 
         logger.debug("Map Origin: {0}".format(self.map_origin))
 
         # Calculate bounding box based on lat/lon
-        min_xy = self._translate_coords(self.bounds['minlat'],
-                                        self.bounds['minlon'])
-        max_xy = self._translate_coords(self.bounds['maxlat'],
-                                        self.bounds['maxlon'])
+        min_xy = self._translate_coords(self.bounds['origin'])
+        max_xy = self._translate_coords(self.bounds['corner'])
         self.bounds['min_xy'] = min_xy
         self.bounds['max_xy'] = max_xy
 
-        self.bounds['size'] = Point(abs(min_xy.x) + abs(max_xy.x),
-                                    abs(min_xy.y) + abs(max_xy.y),
+        self.bounds['size'] = Point(2 * abs(min_xy.x),
+                                    2 * abs(min_xy.y),
                                     0)
 
         logger.debug("Bound box size {0}".format(self.bounds['size']))
         logger.debug(
             "Bound box 'minlat': {0}, 'maxlat': {1}\n\t\t\t\t\t  "
-            "'minlon': {2}, 'maxlon': {3}".format(self.bounds['minlat'],
-                                                  self.bounds['maxlat'],
-                                                  self.bounds['minlon'],
-                                                  self.bounds['maxlon'])
+            "'minlon': {2}, 'maxlon': {3}".format(self.bounds['origin'].lat,
```

Comment:
Great code for a tip: if you define the `__repr__` method of the `LatLon` class then you can express how that object should be converted to text, and hence do something like:
```
"Bounding box from {1} to {2}".format(self.bounds['origin'], self.bounds['corner'])
```